cmd/xl-sets.go

```diff
@@ -1592,15 +1565,8 @@ func (s *xlSets) HealFormat(ctx context.Context, dryRun bool) (res madmin.HealRe
 		// Replace with new reference format.
 		s.format = refFormat
 
-		s.xlDisksMu.Lock()
-		{
-			// Disconnect/relinquish all existing disks.
-			s.xlDisks.Close()
-
-			// Re initialize disks, after saving the new reference format.
-			s.xlDisks = s.reInitDisks(refFormat, storageDisks, tmpNewFormats)
-		}
-		s.xlDisksMu.Unlock()
+		// Disconnect/relinquish all existing disks and allow the reconnect loop to bring back and include all the disks.
```

Comment:
The issue with this change, is that `mc admin heal` won't heal immediately disks. The reformat actually happens, but healing buckets/objects won't happen until disks will reconnect again. I am not sure if it is worth removing reInitDisks()

Reply:
I have added `s.connectDisks()` which would do the same as reinit-disks, can you check now?